### PR TITLE
[JUJU-998] Fix/lp1968643

### PR DESCRIPTION
**Pull request by ycliuhw**
This PR allows `juju_start_unit`, `juju_stop_unit` and `juju_unit_status` returning proper `not supported` errors on CAAS `controller`, `operator` and sidecar `charm` container.

## Checklist

 - [ ] ~Requires a [pylibjuju](https://github.com/juju/python-libjuju) change~
 - [ ] ~Added [integration tests](https://github.com/juju/juju/tree/develop/tests) for the PR~
 - [ ] ~Added or updated [doc.go](https://discourse.jujucharms.com/t/readme-in-packages/451) related to packages changed~
 - [x] Comments answer the question of why design decisions were made

## QA steps

```console
$ juju bootstrap microk8s k1

$ juju add-model t1

$ juju deploy hello-kubecon

$ juju deploy minio

$ mkubectl exec -ncontroller-k1 pod/controller-0 -c api-server  -it -- bash --login
root@controller-0:/var/lib/juju# juju_start_unit hello-kubecon/0
error: start units for {[hello-kubecon/0]} not supported
root@controller-0:/var/lib/juju# juju_stop_unit hello-kubecon/0
error: stop units for {[hello-kubecon/0]} not supported
root@controller-0:/var/lib/juju# juju_unit_status hello-kubecon/0
error: units status not supported

$ mkubectl exec -nt1 pod/hello-kubecon-0   -it -- bash --login
Defaulted container "charm" out of: charm, gosherve, charm-init (init)
root@hello-kubecon-0:/var/lib/juju# juju_start_unit hello-kubecon/0
error: start units for {[hello-kubecon/0]} not supported
root@hello-kubecon-0:/var/lib/juju# juju_stop_unit hello-kubecon/0
error: stop units for {[hello-kubecon/0]} not supported
root@hello-kubecon-0:/var/lib/juju# juju_unit_status hello-kubecon/0
error: units status not supported

$ mkubectl exec -nt1 pod/minio-operator-0   -it -- bash --login
root@minio-operator-0:/var/lib/juju# juju_start_unit minio/0
error: start units for {[minio/0]} not supported
root@minio-operator-0:/var/lib/juju# juju_stop_unit minio/0
error: stop units for {[minio/0]} not supported
root@minio-operator-0:/var/lib/juju# juju_unit_status minio/0
error: units status not supported

```

## Documentation changes

No

## Bug reference

https://bugs.launchpad.net/juju/+bug/1968643

### CODE REVIEW

File: cmd/containeragent/unit/agent.go [hunk 1]
```diff
@@ -318,6 +323,8 @@ func (c *containerUnitAgent) workers() (worker.Worker, error) {
 		NewSocketName:      addons.DefaultIntrospectionSocketName,
 		PrometheusGatherer: c.prometheusRegistry,
 		WorkerFunc:         introspection.NewWorker,
+		Clock:              clock.WallClock,
```
Comment (wallyworld) [inline cmd/containeragent/unit/agent.go:326]: `c.clk`   ?

**Review by wallyworld — APPROVED** [review]
Thank you

File: cmd/containeragent/unit/agent.go (hunk 1 above)
Comment (ycliuhw) [reply cmd/containeragent/unit/agent.go]: ops, right

**Comment by ycliuhw** [comment]
$$merge$$

**Pull request merged**